### PR TITLE
fix: Filter receiver blocks before starting block

### PR DESCRIPTION
**Pull request by morgsmccauley**
As we fetch blocks for a given day, there is possibility for that day to include blocks lesser than the provided `start_block_height`. This PR ensures that those blocks are filtered out.

### CODE REVIEW

File: block-streamer/src/receiver_blocks/receiver_blocks_processor.rs [hunk 1]
```diff
@@ -187,7 +187,9 @@ impl ReceiverBlocksProcessor {
                 }
 
                 for block_height in bitmap_for_day.iter() {
-                    yield block_height;
+                    if block_height > start_block_height {
```
Comment (darunrs) [inline block-streamer/src/receiver_blocks/receiver_blocks_processor.rs:190]: Our [previous](https://github.com/near/queryapi/blob/adaadfc3b2c94ad7bfdf54cd7d9d887b0b55899a/block-streamer/src/delta_lake_client.rs#L250) implementation had `>=`. What's the correct behavior here? It seems to me like >= is what's right.

File: block-streamer/src/receiver_blocks/receiver_blocks_processor.rs [hunk 2]
```diff
@@ -388,11 +416,10 @@ mod tests {
...
             .stream_matching_block_heights(0, "*.someone.near".to_string());
-        tokio::pin!(stream);
-        let mut result_heights = vec![];
-        while let Some(Ok(height)) = stream.next().await {
-            result_heights.push(height);
-        }
-        assert_eq!(result_heights, vec![1, 5, 10, 15, 100, 105]);
+
+        assert_eq!(
+            stream.try_collect::<Vec<u64>>().await.unwrap(),
```
Comment (darunrs) [inline block-streamer/src/receiver_blocks/receiver_blocks_processor.rs:421]: This is very clean!

**Review by darunrs — APPROVED** [review]
Great catch! Thanks for spotting and fixing it!

File: block-streamer/src/receiver_blocks/receiver_blocks_processor.rs (hunk 1 above)
Comment (morgsmccauley) [reply block-streamer/src/receiver_blocks/receiver_blocks_processor.rs:190]: Ah yeah, `>=` is correct 😅

**Pull request merged**